main.tf

```diff
@@ -0,0 +1,39 @@
+provider "google" {
+  project = local.project.projectId
+}
+
+resource "google_firestore_index" "default-db-index" {
+  collection = "composite-index-test-collection"
+
+  for_each = local.indexes
+  dynamic "fields" {
```

Comment:
Seperated the indexes out into `firestore_index_config.tf` file for development simplicity.